slices/reduce.go

```diff
@@ -0,0 +1,25 @@
+package slices
+
+import "golang.org/x/exp/constraints"
+
+func ReduceWith[S ~[]E, E any, A any](s S, acc A, fn func(A, E) A) A {
+	for _, e := range s {
+		acc = fn(acc, e)
+	}
+
+	return acc
+}
+
+func Reduce[S ~[]E, E any, A any](s S, fn func(A, E) A) A {
+	var acc A
+
+	return ReduceWith(s, acc, fn)
+}
+
+type sumable interface {
```

Comment:
Will that not cause an issue? The fact that we have a private constraint on a publicly exposed func?

Reply:
It won't bother the compiler since it is used as a constraint not as a value.